### PR TITLE
Store coercions in alias trackers

### PR DESCRIPTION
**Pull request by lukemaurer**
Since coercions are currently trivial, this doesn't do much yet.

### CODE REVIEW

File: middle_end/flambda/types/env/aliases.ml [hunk 1]
```diff
@@ -16,25 +16,40 @@
 
 module Const = Reg_width_things.Const
 
+type coercion_to_canonical = Coercion.t
+type map_to_canonical = coercion_to_canonical Name.Map.t
```
Comment (poechsel) [inline middle_end/flambda/types/env/aliases.ml]: What about introducing a new module Map_to_canonical here? [map_inter] and [map_union] go directly under it

File: middle_end/flambda/types/env/aliases.ml [hunk 2]
```diff
@@ -45,15 +60,15 @@ module Aliases_of_canonical_element : sig
   val empty : t
   val is_empty : t -> bool
 
-  val add : t -> Name.t -> Name_mode.t -> t
+  val add : t -> Name.t -> coercion_to_canonical:Coercion.t -> Name_mode.t -> t
```
Comment (poechsel) [inline middle_end/flambda/types/env/aliases.ml]: Nitpick: the type could be [coercion_to_canonical] I think.

File: middle_end/flambda/types/env/aliases.ml [hunk 3]
```diff
@@ -733,7 +780,7 @@ let get_aliases t element =
       Aliases_of_canonical_element.all
         (get_aliases_of_canonical_element t ~canonical_element)
     in
-    Alias_set.create ~canonical_element ~alias_names
+    Alias_set.create ~canonical_element ~coercion_from_canonical_to_element:Coercion.id ~alias_names
```
Comment (poechsel) [inline middle_end/flambda/types/env/aliases.ml]: This should fit on 80 columns (there's probably a few other lines that are exceeding this limit in this file)
```suggestion
    Alias_set.create 
      ~canonical_element 
      ~coercion_from_canonical_to_element:Coercion.id 
      ~alias_names
```

Comment (lukemaurer) [reply middle_end/flambda/types/env/aliases.ml]: Fixed (here and in a few other places)

**Pull request merged**